node-support/src/stateless-support.js

```diff
@@ -0,0 +1,171 @@
+    return AnySupport.serialize(obj, true, false, requireJsonType);
+  }
+
+  deserialize(any) {
+    return this.anySupport.deserialize(any);
+  }
+
+}
+
+
+
+module.exports = class StatelessServices {
+
+  constructor() {
+    this.services = {};
+  }
+
+  addService(entity, allEntities) {
+    this.services[entity.serviceName] = new StatelessSupport(entity.root, entity.service, {
+      commandHandlers: entity.commandHandlers      
+    }, allEntities);
+  }
+
+  entityType() {
+    return "cloudstate.function.StatelessFunction";
+  }
+
+  register(server) {
+    const includeDirs = [
+      path.join(__dirname, "..", "proto"),
+      path.join(__dirname, "..", "protoc", "include")
+    ];
+    const packageDefinition = protoLoader.loadSync(path.join("cloudstate", "function.proto"), {
+      includeDirs: includeDirs
+    });
+    const grpcDescriptor = grpc.loadPackageDefinition(packageDefinition);
+
+    const statelessService = grpcDescriptor.cloudstate.function.StatelessFunction.service;
+
+    server.addService(statelessService, {
+      handleUnary: this.handleUnary.bind(this),
+      handleStreamedIn: this.handleStreamedIn.bind(this),
+      handleStreamedOut: this.handleStreamedOut.bind(this),
+      handleStreamed: this.handleStreamed.bind(this),
+    });    
+  }
+
+  handleStreamed(call){
+    call.on("data", data => {
+      if (this.services[data.serviceName] && this.services[data.serviceName].commandHandlers.hasOwnProperty(data.name)) {
+        const userStream = {
+          write: (userData) => {
+            const grpcReturn = this.services[data.serviceName].service.methods[data.name].resolvedResponseType.fromObject(userData);    
+            const requireJsonType =true;
+            call.write({        
+              reply:{
+                payload: AnySupport.serialize(grpcReturn, false, false, requireJsonType)          
+              }        
+            });
+          },
+          end: () => call.end(),                 
+        }
+        // We call this every time and send a way to stream back .. not sure if this is a good way to do things?
+        this.services[data.serviceName].commandHandlers[data.name](userStream, this.services[data.serviceName].deserialize(data.payload)); 
+      }else{
+        console.warn("There is no user function with name: " + data.serviceName + "." + data.name);        
+      }
+    });
+    call.on("end", () => {
+      console.debug("stream ended")
+      //call.end();      
+    });
+  }
+
+  handleStreamedOut(call){
+    const data = call.request;
+    if (this.services[data.serviceName] && this.services[data.serviceName].commandHandlers.hasOwnProperty(data.name)) {
+      const userStream = {
+        write: (userData) => {
+          const grpcReturn = this.services[data.serviceName].service.methods[data.name].resolvedResponseType.fromObject(userData);    
+          const requireJsonType =true;
```

Comment:
Is this always true?

Reply:
To tell you the truth .. I don't think it makes a difference.  In some cases where it can't find the proto it might try falling back to json.